engine/classes/Elgg/Notifications/Event.php

```diff
@@ -67,30 +70,16 @@ public function getActor() {
 	 * @return int
 	 */
 	public function getActorGUID() {
-		return $this->actor_guid;
+		return $this->actor ? $this->actor->guid : 0;
 	}
 
 	/**
 	 * Get the object of the event
 	 *
-	 * @return \ElggData
+	 * @return ElggData
 	 */
 	public function getObject() {
-		switch ($this->object_type) {
```

Comment:
I guess that's the part where we were validating the existence. We can't test this unless we use EntityMocks strategy for relationships and extenders.